kernel/memory_structs/src/lib.rs

```diff
@@ -7,19 +7,35 @@
 
 #![no_std]
 #![feature(step_trait)]
+#![allow(incomplete_features)]
```

Comment:
whoa, what's this? haven't seen this before. Is it related to using enum variants as const generic params?

Reply:
right, it's to suppress the compiler warning `"the feature `adt_const_params` is incomplete and may not be safe to use and/or cause compiler crashes"`